mlflow_faculty/trackingstore.py

```diff
@@ -12,14 +12,36 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from uuid import UUID
 
+import faculty
+from mlflow.entities import ViewType, Experiment, LifecycleStage
+from mlflow.exceptions import MlflowException
 from mlflow.store.abstract_store import AbstractStore
-from mlflow.entities import ViewType
+from six.moves import urllib
+
+from mlflow_faculty.mlflow_converters import (
+    faculty_experiment_to_mlflow_experiment,
+)
 
 
 class FacultyRestStore(AbstractStore):
     def __init__(self, store_uri, **_):
-        pass
+        parsed_uri = urllib.parse.urlparse(store_uri)
+        if parsed_uri.scheme != "faculty":
+            raise ValueError("Not a faculty URI: {}".format(store_uri))
+
+        cleaned_path = parsed_uri.path.strip("/")
+        try:
+            self._project_id = UUID(cleaned_path)
+        except ValueError:
+            raise ValueError(
+                "{} in given URI {} is not a valid UUID".format(
```

Comment:
Maybe we could mention what is the meaning of the UUID, e.g.: "Project ID `{}` in given URI `{}` is not..." (also quotes around the ID as it could be an empty string).
```suggestion
                "Project ID '{}' in given URI {} is not a valid UUID".format(
```